style: enhance history list by adding Time-Based categories

growupanand opened the pull request:
added categories - today, yesterday, thisWeek, thisMonth and older

✅ Closes: #42


<img width="438" alt="image" src="https://github.com/user-attachments/assets/dd82c243-e462-4069-85f6-7e5aab14cde0" />

bhajneet reviewed (changes requested):
Awesome! Some feedback:

- Why was margin removed from lines 75-77 in journey.css? Then you add more margin in line 139. It looks very spaced out and doesn't seem to fit with the overall design theme of the website.
- Full caps titles also don't fit with the design theme of the website. You can take a look at how titles are added to controls options for example.
- The categories should be "today", "yesterday", days of the previous 5 (e.g. from today there would be "thursday", "wednesday", "tuesday", "monday", "sunday"), and the after those 5 it would a full-week and should switch to short dates only (e.g. "21 Jun" or "Jun 21" depending on locale) and onwards. We can remove the "this week" and "this month". And the right side can show just the clock always so it's easy.
- Is the clock timestamp shown on the right side locale dependent? Will it show a 24 hour clock for users that see their time that way?
- We iterate over the items in history twice, once to sort them by timestamp on line 120, then again to categorize them on line 134. I would like to see this done in one iteration instead.
- The reason to originally go from an object on line 120 to an array is that objects don't always hold order in terms of insertion, but arrays do. However reading the documentation it seems that may not be necessary. Is there even a need to sort the items? To quote documentation (correct me if I'm wrong btw): _The keys in Map are ordered while keys added to object are not. Thus, when iterating over it, a Map object returns keys in order of insertion. (Note that in the ECMAScript 2015 spec objects do preserve creation order for string and Symbol keys, so traversal of an object with ie only string keys would yield keys in order of insertion)_

growupanand commented:
> Awesome! Some feedback:
> 
> * Why was margin removed from lines 75-77 in journey.css? Then you add more margin in line 139. It looks very spaced out and doesn't seem to fit with the overall design theme of the website.
> * Full caps titles also don't fit with the design theme of the website. You can take a look at how titles are added to controls options for example.
> * The categories should be "today", "yesterday", days of the previous 5 (e.g. from today there would be "thursday", "wednesday", "tuesday", "monday", "sunday"), and the after those 5 it would a full-week and should switch to short dates only (e.g. "21 Jun" or "Jun 21" depending on locale) and onwards. We can remove the "this week" and "this month". And the right side can show just the clock always so it's easy.
> * Is the clock timestamp shown on the right side locale dependent? Will it show a 24 hour clock for users that see their time that way?
> * We iterate over the items in history twice, once to sort them by timestamp on line 120, then again to categorize them on line 134. I would like to see this done in one iteration instead.
> * The reason to originally go from an object on line 120 to an array is that objects don't always hold order in terms of insertion, but arrays do. However reading the documentation it seems that may not be necessary. Is there even a need to sort the items? To quote documentation (correct me if I'm wrong btw): _The keys in Map are ordered while keys added to object are not. Thus, when iterating over it, a Map object returns keys in order of insertion. (Note that in the ECMAScript 2015 spec objects do preserve creation order for string and Symbol keys, so traversal of an object with ie only string keys would yield keys in order of insertion)_

Thanks for the feedback, please find my comments in same order:

- You're right, I removed the global margin to add targeted spacing between category groups instead. The intention was to create visual separation between different time periods, but I understand it might be too spaced out. Happy to adjust the spacing to better match the overall design theme - would you prefer specific margins or a different approach?
- Good point about the full caps not fitting the design theme. I've removed the text-transform: uppercase and the category titles now use normal case to better align with the overall design language of the site.

- I've updated the category logic to show 'today', 'yesterday', then the previous 5 days by name, followed by short dates for older items. The right side now always shows just the clock time, which will automatically use 24-hour format for users with that locale preference.
   
<img width="326" alt="image" src="https://github.com/user-attachments/assets/8e6d1221-1969-419a-84aa-bbb177b7d941" />

- Yes, the clock timestamp is locale-dependent. I'm using toLocaleTimeString([], { timeStyle: 'short' }) which automatically respects the user's locale settings - it will show 24-hour format for users who have that preference and 12-hour format with AM/PM for others.
- Good point about the double iteration. The sorting step is necessary to maintain chronological order within each category, but I could optimize this by combining both operations. However, the current approach keeps the code readable and the performance impact is minimal since we're dealing with user history (typically small datasets). Would you prefer me to optimize this into a single pass, or is the current readable approach acceptable?
- You're correct that both objects and Maps preserve insertion order in modern JavaScript. However, the sorting is still necessary because we need items ordered by timestamp (newest first), not by insertion order. The history object keys were likely inserted at different times and not necessarily in chronological order, so we need the explicit sort to ensure proper chronological display within each category.


I've addressed each comment above and would be happy to make any additional adjustments. A few thoughts for the future:

- **Design system:** It would be valuable to establish a design system/component library for consistency across UI elements like typography, spacing, colors, and interaction patterns. This would help prevent similar design theme questions in future PRs.
